x-pack/plugins/observability_onboarding/server/saved_objects/observability_onboarding_status.ts

```diff
@@ -22,6 +22,12 @@ export interface SystemLogsState {
   namespace: string;
 }
 
+export interface ElasticAgentStepPayload {
+  agentId: string;
+}
+
+export type StepPayloadType = ElasticAgentStepPayload;
```

Comment:
isn't this redundant. May be i am missing something ?

Reply:
I was preparing for extension in the future, as well as we have it for the different types of logs we could have it for the different steps

Reply:
Duplicate types can become nasty very quickly. Unless you are extending this type, i would avoid

Reply:
Addressed in [202ad51](https://github.com/elastic/kibana/pull/170124/commits/202ad516a7bdaa6e22fbb85735f8988438535163).